From 9821c5863270fe81b93a3fd9c06d642c63fbb59b Mon Sep 17 00:00:00 2001
From: vivekana <vivekana@xilinx.com>
Date: Wed, 8 Sep 2021 13:10:50 -0700
Subject: [PATCH] recipes-utils : jupyterlab server

Start the daemon as root user. Also update path to Notebooks
and global configuration for jupyterlab.
---
 .../start-jupyterlab/jupyter_lab_config.py             |  3 +++
 .../start-jupyterlab/jupyter_notebook_config.py        |  5 -----
 .../start-jupyterlab/jupyterlab-server                 | 10 +++++-----
 3 files changed, 8 insertions(+), 10 deletions(-)
 create mode 100644 recipes-utils/start-jupyterlab/start-jupyterlab/jupyter_lab_config.py
 delete mode 100644 recipes-utils/start-jupyterlab/start-jupyterlab/jupyter_notebook_config.py

diff --git a/jupyter_lab_config.py b/jupyter_lab_config.py
new file mode 100644
index 0000000..6f94bd6
--- /dev/null
+++ b/jupyter_lab_config.py
@@ -0,0 +1,3 @@
+c.ServerApp.port = 8888
+c.ServerApp.iopub_data_rate_limit = 100000000
+c.ServerApp.allow_root=True
diff --git a/jupyter_notebook_config.py b/jupyter_notebook_config.py
deleted file mode 100644
index fdb5cc5..0000000
--- a/jupyter_notebook_config.py
+++ /dev/null
@@ -1,5 +0,0 @@
-c.NotebookApp.port = 8888
-c.NotebookApp.iopub_data_rate_limit = 100000000
-c.NotebookApp.allow_root=True
-c.NotebookApp.open_browser=False
-
diff --git a/jupyterlab-server b/jupyterlab-server
index ef3edd1..da4da0f 100755
--- a/jupyterlab-server
+++ b/jupyterlab-server
@@ -8,10 +8,10 @@
 # Short-Description: Start Jupyter Lab server as petalinux user
 ### END INIT INFO
 
-OWNER="petalinux"
-GROUP="petalinux"
-HOME=`(cd ~petalinux && pwd) || echo 'none'`
-NBDIR="${HOME}/notebooks"
+OWNER="root"
+GROUP="root"
+HOME=`(cd ~root && pwd) || echo 'none'`
+NBDIR="/usr/share/notebooks"
 
 DAEMON_PATH="/usr/bin/jupyter-lab"
 DAEMON_ARGS="--no-browser --notebook-dir=$NBDIR"
@@ -28,7 +28,7 @@ queryip() {
     for i in {1..20}
     do
         echo -n "."
-        ip=`ifconfig eth0 | grep -oP 'inet \d+\.\d+\.\d+\.\d+' | cut -d ' ' -f2`
+        ip=`ifconfig eth0 | grep -oP 'inet addr:\d+\.\d+\.\d+\.\d+' | cut -d ':' -f2`
         [ ! -z "$ip" ] && break
         sleep 2
     done
-- 
2.17.1

